File: 007/problem007.hs
isPrime :: Int -> Int -> Bool
isPrime index num
    | index <= floor (sqrt (fromIntegral num)) = 
        if rem num index == 0 && index /= num then
            False
        else
            isPrime (index + 1) num
    | otherwise = True

getNPrimes :: Int -> Int -> Int -> Int
getNPrimes current count limit 
    | count <= limit =
        if isPrime 2 current then
            getNPrimes (current + 1) (count + 1) limit
        else
            getNPrimes (current + 1) count limit
    | otherwise = current

main = do
    print ((getNPrimes 1 0 10001) - 1)
    

File: 007/problem007.c
#include <stdio.h>

int main(int argc, char *argv[]) {
    int primes[10001] = {0};
    int count = 0;
    int current = 2;

    while (count < 10001) {
        int i = 0;
        int new = 1;
        while (primes[i] != 0) {
            if (current % primes[i] == 0) {
                new = 0;
                break;
            }
            i++;
        }
        if (new) {
            primes[i] = current;
            count++;
        }
        current++;
    }
    printf("%i\n", --current);
    return 0;
}
